diff --git a/lib/Generator.php b/lib/Generator.php
index b578b84..5612b22 100644
--- a/lib/Generator.php
+++ b/lib/Generator.php
@@ -74,9 +74,9 @@ class Generator {
 			}
 
 			if ($this->config->getEnforceSpecialCharacters()) {
-				$password .= $this->random->generate(1, ISecureRandom::CHAR_SYMBOLS);
+				$password .= $this->random->generate(1, str_replace('\\', '', ISecureRandom::CHAR_SYMBOLS));
 				$length -= 1;
-				$chars .= ISecureRandom::CHAR_SYMBOLS;
+				$chars .= str_replace('\\', '', ISecureRandom::CHAR_SYMBOLS);
 			}
 
 			if ($chars === '') {
